src/array/cuda/spmm.cu

```diff
@@ -400,6 +501,117 @@ void SpMMCsr(const std::string& op, const std::string& reduce,
   }
 }
 
+/*!
+ * \brief CUDA implementation of g-SpMM on Csr format.
+ * \note use cusparse if the reduce operator is `sum` and there is
+ *       no broadcast, use dgl's kernel in other cases.
+ */
+template <int XPU, typename IdType, int bits>
+void SpMMCsrHetero(const std::string& op, const std::string& reduce,
+             const BcastOff& bcast,
+             const std::vector<CSRMatrix>& vec_csr,
+             const std::vector<NDArray>& vec_ufeat,
+             const std::vector<NDArray>& vec_efeat,
+             std::vector<NDArray> vec_out,
+             const std::vector<NDArray>& out_aux,
+             const std::vector<dgl_type_t>& ufeat_eid,
+             const std::vector<dgl_type_t>& out_eid) {
+  int64_t feat_len = bcast.out_len;
+  bool is_scalar_efeat = vec_efeat.size() != 0;
+  bool use_efeat = op != "copy_lhs";
+  // TODO(Israt): 1:Resolve PR-https://github.com/dmlc/dgl/issues/2995
+  // to use maxstream > 1
+  int maxstrm = 1;
+  std::vector<cudaStream_t> stream(maxstrm);
+  for (int i = 0; i < maxstrm; i++) {
+    CUDA_CALL(cudaStreamCreate(&(stream[i])));
```

Comment:
We still shouldn't create a stream here, as we could run into a race condition with the default stream (technically I think the default stream should block around it by default, but I'd consider it dangerous). I think for now, it would be better to do:
```
const int maxstrm = 1;
std::vector<cudaStream_t> stream{static_cast<cudaStream_t>(0)};
```
Such that everything will be run on the default stream.